Documentation/dotnet-mage/ApplicationDeployment.cs

```diff
@@ -0,0 +1,129 @@
+﻿using System;
+
+namespace ClickOnceHelper
+{
+    public class ApplicationDeployment
+    {
+        private static ApplicationDeployment currentDeployment = null;
+        private static bool currentDeploymentInitialized = false;
+
+        private static bool isNetworkDeployed = false;
+        private static bool isNetworkDeployedInitialized = false;
+
+        public static bool IsNetworkDeployed
+        {
+            get
+            {
+                if (!isNetworkDeployedInitialized)
+                {
+                    ApplicationDeployment.isNetworkDeployed = false;
+
+                    string value = Environment.GetEnvironmentVariable("ClickOnce_IsNetworkDeployed");
+                    if (value?.ToLower() == "true")
```

Comment:
consider using bool.TryParse here like you did for IsFirstRun

Reply:
Thanks @John-Hart - I've updated parsing in the sample class with latest commit.